packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -1,73 +1,166 @@
-/**
- * <TextInput
-  label="Enter Name"
-  labelPosition="left"
-  placeholder="Enter your name"
-  icon={UserIcon}
-  showClearButton
-  helpText="Enter first name and last name. Eg: Kamlesh Chandnani"
-  errorText="Name cannot be blank"
-  successText="Name validated"
-  validationState={inputValue.length < 10 ? 'error' : 'success'}
-/>
- */
-
+import React from 'react';
 import type { ReactElement } from 'react';
 import { StyledBaseInput } from './StyledBaseInput';
+import Box from '~components/Box';
+import { FormHint, FormLabel } from '~components/Form';
+import { getPlatformType } from '~utils';
+import type { FormLabelProps } from '~components/Form/FormLabel';
+import type { FormHintProps } from '~components/Form/FormHint';
+import { useFormId } from '~components/Form/useFormId';
 
-type HandleOnChange = ({
-  event,
+export type HandleOnChange = ({
   inputName,
-  value,
+  inputValue,
 }: {
-  event?: React.ChangeEvent<HTMLInputElement>;
   inputName?: string;
-  value?: string;
+  inputValue?: React.ChangeEvent<HTMLInputElement> | string;
 }) => void;
 
-type OnChange = ({
-  event,
-  name,
-  value,
-}: {
-  event?: React.ChangeEvent<HTMLInputElement>;
-  name?: string;
-  value?: string;
-}) => void;
+export type OnChange = ({ name, value }: { name?: string; value?: string }) => void;
```

Comment:
event?

Reply:
don't need event in case of input field. just sending out name and value to the consumer. also event is diff on react native and essentially there's no use as of now